closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -344,23 +344,29 @@ private static void getProfileID(Properties props) {
                         + selectedProfile);
             }
             String defaultMatch = null;
+            Boolean profileExist = false;
             for (Object keyObject : props.keySet()) {
                 if (keyObject instanceof String key) {
-                    if (key.startsWith(potentialProfileID) && key.endsWith(".desc.default")) {
-                        // Check if property is set to true.
-                        if (Boolean.parseBoolean(props.getProperty(key))) {
-                            // Check if multiple defaults exist and act accordingly.
-                            if (defaultMatch == null) {
-                                defaultMatch = key.split("\\.desc")[0];
-                            } else {
-                                printStackTraceAndExit("Multiple default RestrictedSecurity"
-                                        + " profiles for " + selectedProfile);
+                    if (key.startsWith(potentialProfileID)) {
```

Comment:
This is not right: It will match when it should not. For example, suppose `key` is "sample2", desired profile is "sample".

This also complains when it should not. Suppose there's only one profile with a name beginning with "sample": it should not be necessary to have the `sample.desc.default` property.

Reply:
@keithc-ca 
Yes, we found this problem weeks ago, and I already opened the issue 2 weeks ago to address this, link https://github.ibm.com/runtimes/jit-crypto/issues/573. I am planning to open another PR to fix this problem. So, do you think, I fix this problem in the current PR, or as I planned, open another PR to fix this?

Reply:
Issues relating to public repositories should be public as well (unless there's a good reason otherwise - e.g. a security vulnerability). Please move that issue to https://github.com/eclipse-openj9/openj9/issues.

Reply:
@keithc-ca The issue created under the openj9 repo, https://github.com/eclipse-openj9/openj9/issues/20518. I will open another PR to address this issue.